### PR TITLE
Issue 331 migrate json language catalogs to po

### PR DESCRIPTION
**Pull request by bram-atmire**

### CODE REVIEW

File: src/app/shared/lang-switch/lang-switch.component.spec.ts [hunk 1]
```diff
@@ -27,7 +27,10 @@ class CustomLoader implements TranslateLoader {
 }
 /* tslint:enable:quotemark */
 /* tslint:enable:object-literal-key-quotes */
+<<<<<<< HEAD
+=======
 
+>>>>>>> 02e0b879... Resolving TSLint issues
```
Comment (paulo-graca) [inline src/app/shared/lang-switch/lang-switch.component.spec.ts]: Please remove this.

File: src/app/shared/log-in/log-in.component.html [hunk 2]
```diff
@@ -1,28 +1,28 @@
...
          class="form-control form-control-lg position-relative mb-3"
-         placeholder="{{'login.form.password' | translate}}"
+         placeholder="{{'Password' | translate}}"
          formControlName="password"
          required
          type="password">
   <div *ngIf="(error | async) && hasError" class="alert alert-danger" role="alert" @fadeOut>{{ (error | async) | translate }}</div>
   <div *ngIf="(message | async) && hasMessage" class="alert alert-info" role="alert" @fadeOut>{{ (message | async) | translate }}</div>
-  <button class="btn btn-lg btn-primary btn-block mt-3" type="submit" [disabled]="!form.valid">{{"login.form.submit" | translate}}</button>
+  <button class="btn btn-lg btn-primary btn-block mt-3" type="submit" [disabled]="!form.valid">{{"login.Submit" | translate}}</button>
```
Comment (artlowel) [inline src/app/shared/log-in/log-in.component.html]: `login.Submit` here seems like a mistake

File: src/app/+browse-by/+browse-by-title-page/browse-by-title-page.component.html [hunk 3]
```diff
@@ -1,7 +1,7 @@
 <div class="container">
   <div class="browse-by-title w-100 row">
     <ds-browse-by class="col-xs-12 w-100"
-      title="{{'browse.title' | translate:{collection: '', field: 'Title', value: ''} }}"
+      title="{{'Browsing {{ collection }} by {{ field }} {{ value }}' | translate:{collection: '', field: 'Title', value: ''} }}"
```
Comment (artlowel) [inline src/app/+browse-by/+browse-by-title-page/browse-by-title-page.component.html]: I get the exception below on this label. That likely means all the other identical labels in other components will cause the same exception:
```
Parser Error: Unexpected token Lexer Error: Unterminated quote at column 24 in expression ['Browsing {{ collection ] at column 25 in [{{'Browsing {{ collection }} by {{ field }} {{ value }}' | translate:{collection: '', field: 'Title', value: ''} }}] in ng:///BrowseByModule/BrowseByTitlePageComponent.html@3:6
```

Comment (bram-atmire) [reply src/app/+browse-by/+browse-by-title-page/browse-by-title-page.component.html]: Whenever there's the two curly bracket sequence, I've been breaking it into separate strings cfr https://stackoverflow.com/questions/16868024/how-do-i-escape-curly-braces-for-display-on-page-when-using-angularjs

For example, footer component or pagination component htmls.

**Comment by bram-atmire** [comment]
@artlowel @paulo-graca 
Normally the build should now work and the tests should pass. I also made another rebase and made sure that both dspace.pot (the template file), as well as en.po (the English translations) are up to date with the latest keys in en.json.

To help with the transition, I made sure that:
- the original key name/path from en.json is available in .po and .pot comments
- the ORDER of the keys corresponds with the order in en.json to the extent possible.

Can we identify any remaining blockers for stopping adding new keys to en.json? Would really love to see if we can finalize the transition, so we can start on documentation and onboarding of translators and developers with the new way of working with translations.

Dipped into that "Poedit" software for managing the translations, and it really looks like a great tool that will make life a lot easier for translators.

File: src/app/shared/log-out/log-out.component.html [hunk 4]
```diff
@@ -3,5 +3,5 @@
 
   <div *ngIf="(error | async) && hasError" class="alert alert-danger" role="alert" @fadeOut>{{ error | async }}</div>
 
-  <button class="btn btn-lg btn-primary btn-block mt-3" (click)="logOut()">{{"logout.form.submit" | translate}}</button>
+  <button class="btn btn-lg btn-primary btn-block mt-3" (click)="logOut()">{{"logout.Submit" | translate}}</button>
```
Comment (AlexanderS) [inline src/app/shared/log-out/log-out.component.html]: `logout.` seems to be extra here.

File: src/app/shared/search-form/search-form.component.html [hunk 5]
```diff
@@ -1,15 +1,15 @@
 <form #form="ngForm" (ngSubmit)="onSubmit(form.value)" class="row" action="/search">
 <div *ngIf="isNotEmpty(scopes)" class="col-12 col-sm-3">
   <select [(ngModel)]="scope" name="scope" class="form-control" aria-label="Search scope" (change)="onScopeChange($event.target.value)">
-    <option value>{{'search.form.search_dspace' | translate}}</option>
-    <option *ngFor="let scopeOption of scopes" [value]="scopeOption.id">{{scopeOption?.name ? scopeOption.name : 'search.form.search_dspace' | translate}}</option>
+    <option value>{{'Search DSpace' | translate}}</option>
+    <option *ngFor="let scopeOption of scopes" [value]="scopeOption.id">{{scopeOption?.name ? scopeOption.name : 'search.Search_dspace' | translate}}</option>
```
Comment (AlexanderS) [inline src/app/shared/search-form/search-form.component.html]: `search.` seems to be extra here. Maybe it should be `'Search DSpace'` here, too.

File: src/app/shared/form/form.component.spec.ts [hunk 6]
```diff
@@ -345,7 +345,7 @@ describe('FormComponent test suite', () => {
       });
     });
 
-    it('should emit submit Event on form submit whether the form is valid', () => {
+    it('should emit submit Event on Submit whether the form is valid', () => {
```
Comment (AlexanderS) [inline src/app/shared/form/form.component.spec.ts]: This maybe is a search/replace error.

File: src/app/shared/form/form.component.spec.ts [hunk 7]
```diff
@@ -359,7 +359,7 @@ describe('FormComponent test suite', () => {
       expect(formComp.submitForm.emit).toHaveBeenCalled();
     });
 
-    it('should not emit submit Event on form submit whether the form is not valid', () => {
+    it('should not emit submit Event on Submit whether the form is not valid', () => {
```
Comment (AlexanderS) [inline src/app/shared/form/form.component.spec.ts]: This maybe is a search/replace error, too.

File: src/app/shared/form/form.component.html [hunk 8]
```diff
@@ -50,9 +50,9 @@
       <div class="form-group row">
 
         <div class="col text-right">
-          <button type="reset" class="btn btn-default" (click)="reset()">{{'form.cancel' | translate}}</button>
-          <button type="submit" class="btn btn-primary" (click)="onSubmit()"
-                  [disabled]="!(isValid() | async)">{{'form.submit' | translate}}
+          <button type="reset" class="btn btn-default" (click)="reset()">{{'Cancel' | translate}}</button>
+          <button type="submit" class="btn btn-primary" (click)="onSubmit($event)"
```
Comment (AlexanderS) [inline src/app/shared/form/form.component.html]: Any reason why `$event` is included here now?

File: src/app/shared/form/builder/ds-dynamic-form-ui/models/relation-group/dynamic-relation-group.component.html [hunk 9]
```diff
@@ -1,14 +1,14 @@
 <a *ngIf="!(formCollapsed | async)"
    class="close position-relative"
-   ngbTooltip="{{'form.group-collapse-help' | translate}}"
+   ngbTooltip="{{'Collapse-help' | translate}}"
```
Comment (AlexanderS) [inline src/app/shared/form/builder/ds-dynamic-form-ui/models/relation-group/dynamic-relation-group.component.html]: This should be `{{'Click here to collapse' | translate}}`.

File: src/app/shared/form/builder/ds-dynamic-form-ui/models/relation-group/dynamic-relation-group.component.html [hunk 10]
```diff
@@ -1,14 +1,14 @@
...
    placement="left">
   <span class="fas fa-angle-up fa-fw fa-2x"
         aria-hidden="true"
         (click)="collapseForm()"></span>
 </a>
 <a *ngIf="(formCollapsed | async)"
    class="close position-relative"
-   ngbTooltip="{{'form.group-expand-help' | translate}}"
+   ngbTooltip="{{'Expand-help' | translate}}"
```
Comment (AlexanderS) [inline src/app/shared/form/builder/ds-dynamic-form-ui/models/relation-group/dynamic-relation-group.component.html]: This should be `{{'Click here to expand and add more elements' | translate}}`.

File: src/app/shared/pagination/pagination.component.html [hunk 11]
```diff
@@ -2,17 +2,17 @@
...
-            <h6 class="dropdown-header">{{ 'pagination.sort-direction' | translate}}</h6>
-            <button class="dropdown-item" *ngFor="let direction of (sortDirections | dsKeys)" (click)="doSortDirectionChange(direction.value)"><i [ngClass]="{'invisible': direction.value !== sortDirection}" class="fas fa-check" aria-hidden="true"></i> {{'sorting.' + direction.key | translate}} </button>
+            <h6 class="dropdown-header">{{ 'Results Per Page' | translate}}</h6>
+            <button class="dropdown-item" *ngFor="let item of pageSizeOptions" (click)="doPageSizeChange(item)"><i [ngClass]="{'invisible': item != pageSize}" class="fa fa-check" aria-hidden="true"></i> {{item}} </button>
+            <h6 class="dropdown-header">{{ 'Sort Options' | translate}}</h6>
+            <button class="dropdown-item" *ngFor="let direction of (sortDirections | dsKeys)" (click)="doSortDirectionChange(direction.value)"><i [ngClass]="{'invisible': direction.value !== sortDirection}" class="fa fa-check" aria-hidden="true"></i> {{'sorting.' + direction.key | translate}} </button>
```
Comment (AlexanderS) [inline src/app/shared/pagination/pagination.component.html]: Was the change from `fas` to `fa` intended?

File: src/app/shared/view-mode-switch/view-mode-switch.component.html [hunk 12]
```diff
@@ -6,7 +6,7 @@
        routerLinkActive="active"
        [class.active]="currentMode === viewModeEnum.List"
        class="btn btn-secondary">
-          <i class="fas fa-list" title="{{'search.view-switch.show-list' | translate}}"></i>
+          <i class="fa fa-list" title="{{'Show as list' | translate}}"></i>
```
Comment (AlexanderS) [inline src/app/shared/view-mode-switch/view-mode-switch.component.html]: Also a change from `fas` to `fa`.

File: src/app/shared/view-mode-switch/view-mode-switch.component.html [hunk 13]
```diff
@@ -15,6 +15,6 @@
        routerLinkActive="active"
        [class.active]="currentMode !== viewModeEnum.List"
        class="btn btn-secondary">
-          <i class="fas fa-th-large" title="{{'search.view-switch.show-grid' | translate}}"></i>
+          <i class="fa fa-th-large" title="{{'Show as grid' | translate}}"></i>
```
Comment (AlexanderS) [inline src/app/shared/view-mode-switch/view-mode-switch.component.html]: And another change from `fas` to `fa`.

File: src/app/shared/log-out/log-out.component.html (hunk 4 above)
Comment (bram-atmire) [reply src/app/shared/log-out/log-out.component.html]: tnx, fixed today

File: src/app/shared/search-form/search-form.component.html (hunk 5 above)
Comment (bram-atmire) [reply src/app/shared/search-form/search-form.component.html]: thanks, fixed today

File: src/app/shared/form/form.component.spec.ts (hunk 6 above)
Comment (bram-atmire) [reply src/app/shared/form/form.component.spec.ts]: thanks, fixed today

File: src/app/shared/form/form.component.spec.ts (hunk 7 above)
Comment (bram-atmire) [reply src/app/shared/form/form.component.spec.ts]: thanks, fixed today

File: src/app/shared/form/form.component.html (hunk 8 above)
Comment (bram-atmire) [reply src/app/shared/form/form.component.html]: no idea how that got in there. thanks, fixed today

File: src/app/shared/form/builder/ds-dynamic-form-ui/models/relation-group/dynamic-relation-group.component.html (hunk 9 above)
Comment (bram-atmire) [reply src/app/shared/form/builder/ds-dynamic-form-ui/models/relation-group/dynamic-relation-group.component.html]: thanks, fixed today

File: src/app/shared/form/builder/ds-dynamic-form-ui/models/relation-group/dynamic-relation-group.component.html (hunk 10 above)
Comment (bram-atmire) [reply src/app/shared/form/builder/ds-dynamic-form-ui/models/relation-group/dynamic-relation-group.component.html]: thanks, fixed today

File: src/app/shared/pagination/pagination.component.html (hunk 11 above)
Comment (bram-atmire) [reply src/app/shared/pagination/pagination.component.html]: Unclear how that replace happened, reverted today.

File: src/app/shared/view-mode-switch/view-mode-switch.component.html (hunk 12 above)
Comment (bram-atmire) [reply src/app/shared/view-mode-switch/view-mode-switch.component.html]: Unclear how that replace happened, reverted today.

File: src/app/shared/view-mode-switch/view-mode-switch.component.html (hunk 13 above)
Comment (bram-atmire) [reply src/app/shared/view-mode-switch/view-mode-switch.component.html]: Unclear how that replace happened, reverted today.

**Comment by AlexanderS** [comment]
@bram-atmire Thanks for fixing my comments. Just a note: The current commit message looks a bit strange.

@tdonohue We could check if DSpace would qualify for the free open source tier of Transifex: https://docs.transifex.com/projects/open-source-project This would make it really easy to contribute new translations (there is a nice web editor).

**Comment by bram-atmire** [comment]
We still need to find a solution for dynamic built keys like like:

type.toLowerCase() + '.page.titleprefix'
item.edit.tabs.status.buttons.' + operation.operationKey + '.label	
item.edit.tabs.status.buttons.' + operation.operationKey + '.button	
item.edit.tabs.status.buttons.' + operation.operationKey + '.button	
item.edit.tabs.' + page + '.head

**Comment by artlowel** [comment]
Because [the translation loader this PR is built on](https://github.com/biesbjerg/ngx-translate-po-http-loader) isn't actively maintained, has no support for server side rendering, and [because the main developer behind ngx-translate isn't quite sure how much support he can give the project in the future](https://twitter.com/OCombe/status/1148274119609335813), we decided in [the meeting of 2019-07-05](https://wiki.duraspace.org/display/DSPACE/2019-07-25+DSpace+7+Working+Group+Meeting) to close this PR, and keep working with ngx-translate's native format for the time being.

To make translating easier for non-developers we'll create a PR to flatten the existing json files. We'll also look in to ways of converting the json files to po or xliff format and back, so those formats can still be used by translators, just not live in the application.

**Comment by tdonohue** [comment]
This PR is currently replaced by #439.  More thoughts/opinions welcome there.

**Pull request closed without merging**